walk-in-interview/src/main/webapp/new-job/script.js

```diff
@@ -179,43 +202,104 @@ function getJobDetailsFromUserInput() {
   const payMax = document.getElementById('pay-max').valueAsNumber;
 
   const requirementsCheckboxes =
-      document.getElementsByName('requirements-list');
+    document.getElementsByName('requirements-list');
   const requirementsList = [];
   requirementsCheckboxes.forEach(({checked, id}) => {
-      if (checked) {
+    if (checked) {
       requirementsList.push(id);
-      }
+    }
   });
 
   const expiry = document.getElementById('expiry').valueAsNumber;
   const duration = document.getElementById('duration').value;
 
   const jobDetails = {
-      jobTitle: name,
-      jobLocation: {
+    jobTitle: name,
+    jobLocation: {
       address: address,
       postalCode: postalCode,
+      region: findRegion(postalCode),
       lat: 1.3039, // TODO(issue/13): get these from places api
       lon: 103.8358,
-      },
-      jobDescription: description,
-      jobPay: {
+    },
+    jobDescription: description,
+    jobPay: {
       paymentFrequency: payFrequency,
       min: payMin,
       max: payMax,
-      },
-      requirements: requirementsList,
-      postExpiryTimestamp: expiry,
-      jobDuration: duration,
+      annualMax: calculateAnnualMax(payMax, payFrequency),
+    },
+    requirements: requirementsList,
+    postExpiryTimestamp: expiry,
+    jobDuration: duration,
   };
 
   return jobDetails;
 }
 
 /**
-* Validates the user input
-* @return {boolean} depending on whether the input is valid or not.
-*/
+ * Returns the Singapore region based on the postal code.
+ * This is done by taking the first two digits of the postal
+ * code and seeing which region it corresponds to.
+ * Central: 01-45
+ * West:  58-71
+ * East: 46-52, 81
+ * North: 72-73, 75-80
+ * North-East: 53-57, 82
+ * @param {String} postalCode The Singapore postal code.
+ * @return {String} The Singapore region.
+ */
+function findRegion(postalCode) {
+  /*
+   * We have made the assumption that if the user's
+   * postal code region is 01, they would have written
+   * 01xxx rather than 1xxx.
+   */
+  const digits = parseInt(postalCode.substring(0, 2));
+
+  if (digits >= 1 && digits <= 45) {
+    return 'CENTRAL';
+  } else if (digits >= 58 && digits <= 71) {
+    return 'WEST';
+  } else if ((digits >= 46 && digits <= 52) ||
+      digits === 81) {
+    return 'EAST';
+  } else if ((digits >= 72 && digits <= 73) ||
+      (digits >= 75 && digits <= 80)) {
+    return 'NORTH';
+  } else if ((digits >= 53 && digits <= 57) ||
+      digits === 82) {
+    return 'NORTH_EAST';
+  }
+  setErrorMessageAndField(/* errorFieldId= */ 'postal-code',
```

Comment:
i think the error message should be set by whoever calls findRegion instead of findRegion doing that itself. findRegion is not aware of the html fields (and rightly so).